aws/resource_aws_cloudwatch_event_bus_policy_test.go

```diff
@@ -0,0 +1,98 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	events "github.com/aws/aws-sdk-go/service/cloudwatchevents"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccAWSCloudwatchEventBusPolicy_basic(t *testing.T) {
+	resourceName := "aws_cloudwatch_event_bus_policy.test"
+	rstring := acctest.RandString(5)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCloudWatchEventBusDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCloudwatchEventBusPolicyConfig(rstring),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSCloudwatchEventBusPolicyExists(resourceName),
+				),
```

Comment:
Could you check for policy properties like statement_id and principal? e.g. check whether policy matches what's being created

Reply:
This now does a deep diff from TF state and what's available in the bus. It does the job and run through the entire logic IMHO and state.

https://github.com/hashicorp/terraform-provider-aws/pull/16874/files?file-filters%5B%5D=.go#diff-23812e233d355579267c55e833a81e4aa3a286298ee1477b978cbcd22ec7ad0cR141

Deferring to Hashicorp core team if they want to be nitpick and check whether X key has changed.